icu4c/source/test/intltest/units_data_test.cpp

```diff
@@ -140,18 +141,22 @@ void UnitsDataTest::testGetPreferencesFor() {
 
     for (const auto &t : testCases) {
         logln(t.name);
-        const UnitPreference *const *prefs;
-        int32_t prefsCount;
-        preferences.getPreferencesFor(t.category, t.usage, t.region, prefs, prefsCount, status);
+        CharString localeID;
+        if (uprv_isASCIILetter(t.region[0]) && uprv_strcmp(t.region, "default") != 0) {
+            localeID.append("und-", status);
+        }
+        localeID.append(t.region, status);
+        Locale locale(localeID.data());
```

Comment:
Same problems as in the Java test. See my comments there.

Reply:
done